src/Util/FileHasher.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Util;
+
+use Rector\Core\Exception\ShouldNotHappenException;
+
+final class FileHasher
+{
+    /**
+     * cryptographic insecure hasing of a string
+     */
+    public function hash(string $string): string
+    {
+        return hash($this->getAlgo(), $string);
+    }
+
+    /**
+     * cryptographic insecure hasing of files
+     *
+     * @param string[] $files
+     */
+    public function hashFiles(array $files): string
+    {
+        $configHash = '';
+        $algo = $this->getAlgo();
+        foreach ($files as $file) {
+            $hash = hash_file($algo, $file);
+            if ($hash === false) {
+                throw new ShouldNotHappenException(sprintf('File %s is not readable', $file));
+            }
+
+            $configHash .= $hash;
+        }
+
+        return $configHash;
+    }
+
+    private function getAlgo(): string
```

Comment:
can be set once in a property in the constructor, not likely to change during execution

Reply:
I prefer a local method over a class wide property for such hardcoded values

Reply:
"hardcoded logic" that is called every time `hash` and `hashFiles` are called, while not necessary.
for a heavier logic a cached property would have been used I guess to avoid repeating the logic.
but here the method is light so it's not about caching, more about "state that the is one and only one algo that will be used by this service, it's not dynamically fetch at runtime depending on the context".
but maybe you'd appreciate more a constant? see https://3v4l.org/9KZlf
in this case it could be a private constant

Reply:
I think this is good enough :+1: